### DISCUSSION TITLE
Multiple responses for a request

### DISCUSSION THREAD

**Discussion by Doomsta** [nats-io/nats.go#1600] [Q&A] [answered]
I just discovered there is a `request_multi` fn in the rust client:
```rust
// With multiple responses.
for msg in nc.request_multi("foo", "Help")?.iter() {}
```
https://docs.rs/nats/latest/nats/#requestresponse

Is there a recommended way to achieve the same behavior in go?

**Comment by Jarema** [comment, marked as answer]
`Request` is using `Publish` and `Subscribe` under the hood.

You can achieve the same goal creating a subscription first:

```go
// with callback
nc.Subscribe("service", cb) 
// or a sync variant
 nc.SubscribeSync("service")
 ```
 
 and then send request:
 ```go
 nc.PublishRequest("requestTo", "service", []byte("data"))
 ```

**Reply by Doomsta** [reply]
Thanks, works for me.
